provisioner/ibm-s3fs-provisioner.go

```diff
@@ -246,11 +247,21 @@ func (p *IBMS3fsProvisioner) validateAnnotations(ctx context.Context, options co
 		}
 	}
 
-	if pvc.SecretNamespace == "" {
-		if sc.SecretNamespace != "" {
-			pvc.SecretNamespace = sc.SecretNamespace
-		} else {
-			pvc.SecretNamespace = options.PVC.Namespace
+	contextLogger.Info(pvcName + ":" + clusterID + " AllowCrossNsSecret: " + strconv.FormatBool(*AllowCrossNsSecret))
+
+	if AllowCrossNsSecret != nil && *AllowCrossNsSecret == false {
+		contextLogger.Info(pvcName + ":" + clusterID + " AllowCrossNsSecret is set to false, the secret will be looked for in same namespace where pvc is created")
+		if pvc.SecretNamespace != "" {
+			contextLogger.Warn(pvcName + ":" + clusterID + " ibm.io/secret-namespace annotation if set will be ignored as AllowCrossNsSecret is set to false")
```

Comment:
`ibm.io/secret-namespace annotation if set will be ignored as AllowCrossNsSecret is set to false` ---> `Ignoring 'ibm.io/secret-namespace' annotation as AllowCrossNsSecret is set to false`